src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -29,6 +32,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        user = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
+        return user;
```

Comment:
```suggestion
        userService.add(user);
        shoppingCartService.registerNewShoppingCart(user);
        return user;
```
Should work this way too